kivakit-core/src/main/java/com/telenav/kivakit/core/thread/StateWatcher.java

```diff
@@ -70,21 +73,21 @@
     @LexakaiJavadoc(complete = true)
     private class Waiter
     {
-        /** The predicate that must be satisfied */
-        final Predicate<State> predicate;
-
-        /** The condition variable to wait on and signal */
-        final Condition condition;
-
         private Waiter(Predicate<State> predicate, Condition condition)
         {
             this.predicate = predicate;
             this.condition = condition;
         }
+
+        /** The predicate that must be satisfied */
+        final Predicate<State> predicate;
+
+        /** The condition variable to wait on and signal */
+        final Condition condition;
     }
 
     /** The most recently reported state */
-    private State current;
+    private volatile State current;
```

Comment:
Only read or written from constructor or when holding a lock -> does not need `volatile`.

Reply:
it's holding a lock, but not a lock on that object. i found that this fixed a bug.

Reply:
Going to skip this.